tools/test-tools/bin/assign-test-ports

```diff
@@ -1,25 +1,35 @@
 #!/usr/bin/env node
 const mod = require("../dist/assignTestPorts.js");
+const nconf = require("nconf");
 
-const firstArg = process.argv[2];
-
-if (firstArg === "--help" || firstArg === "-h" || (firstArg !== undefined && !Number.isInteger(firstArg)) ) {
-	console.log(
-		"Usage: assign-test-ports [initialPort]\n\n" +
-		"Assigns a port to each package in the current pnpm workspace so they can run jest/puppeteer\n" +
-		"tests concurrently without port conflicts, and writes the port mapping to a file.\n\n" +
-		"The initial port is optional and defaults to 9000.\n\n" +
-		"The port mapping file is written to the OS' temporary storage folder (e.g. /tmp/testportmap.json)."
-	);
-	process.exit(0);
-}
+nconf
+	.argv(
+		{
+			"package-manager": {
+				type: "string",
+				describe: "Package manager used for the monorepo.",
+				default: "pnpm",
+				choices: ["pnpm", "lerna"],
+			},
+		},
+		"Usage: assign-test-ports [initialPort] [--package-manager <pnpm | lerna>]\n\n" +
+			"Assigns a port to each package in the current monorepo so they can run jest/puppeteer\n" +
+			"tests concurrently without port conflicts, and writes the port mapping to a file.\n\n" +
+			"The initial port is optional and defaults to 9000.\n\n" +
+			"The port mapping file is written to the OS' temporary storage folder (e.g. /tmp/testportmap.json).",
+	)
+	.env();
 
 // We used to hardcode port 8081 as the initial one
 // but as of 2024-11-25 port 8084 is used by something in the build agent image.
 // If a package that has jest tests ends up assigned that port,
 // jest will fail to start.
 // So try to use a port range where nothing will be listening.
-const initialPortString = firstArg ?? "9000";
+// Note '_' is used for positional arguments in nconf.
+const initialPort = nconf.get("_")[0] ?? 9000;
```

Comment:
I'd expect `nconf.get()` to return a string. I _think_ that gets handled correctly all the way because the first thing we end up doing with it is `<value>++`, but I think it'd be better to explicitly convert to number here.